src/pkg/packager/actions.go

```diff
@@ -0,0 +1,165 @@
+		spinner := message.NewProgressSpinner("Running command \"%s\"", a.Cmd)
+		defer spinner.Success()
+
+		var (
+			ctx    context.Context
+			cancel context.CancelFunc
+			cmd    string
+			err    error
+		)
+
+		cfg := actionGetCfg(actionSet, a)
+
+		if cmd, err = actionCmdMutation(a.Cmd); err != nil {
+			spinner.Errorf(err, "Error mutating command: %s", cmd)
+		}
+
+		// Otherwise, run the command with a timeout handler
+		spinner.Updatef("Waiting for command \"%s\" (timeout: %d seconds)", cmd, cfg.MaxTotalSeconds)
+
+		duration := time.Duration(cfg.MaxTotalSeconds) * time.Second
+		timeout := time.After(duration)
+
+		// Keep trying until the max retries is reached
+		for remaining := cfg.MaxRetries + 1; remaining > 0; remaining-- {
+
+			// If no timeout is set, run the command and return
+			if cfg.MaxTotalSeconds < 1 {
+				spinner.Updatef("Waiting for command \"%s\" (no timeout)", cmd)
+
+				// Try running the command and return if it fails
+				if err := actionRun(context.TODO(), cfg, cmd); err != nil {
+					return fmt.Errorf("command \"%s\" failed: %w", cmd, err)
+				}
+
+				// If the command ran successfully, continue to the next action
+				continue ACTION
+			}
+
+			select {
+			// On timeout abort
+			case <-timeout:
+				cancel()
+				return fmt.Errorf("command \"%s\" timed out", cmd)
+
+			// Otherwise try running the command
+			default:
+				ctx, cancel = context.WithTimeout(context.Background(), duration)
+				defer cancel()
+
+				// Try running the command and continue the retry loop if it fails
+				if err := actionRun(ctx, cfg, cmd); err != nil {
+					message.Debug(err)
+					continue
+				}
+
+				// If the command ran successfully, continue to the next action
+				continue ACTION
+			}
+		}
+
+		// If we've reached this point, the retry limit has been reached
+		return fmt.Errorf("command \"%s\" failed after %d retries", cmd, cfg.MaxRetries)
+	}
+
+	// If we've reached this point, all actions have been run successfully
+	return nil
+}
+
+// Perform some basic string mutations to make commands more useful
+func actionCmdMutation(cmd string) (string, error) {
+	binaryPath, err := os.Executable()
+	if err != nil {
+		return cmd, err
+	}
+
+	// Try to patch the zarf binary path in case the name isn't exactly "./zarf"
+	cmd = strings.ReplaceAll(cmd, "./zarf ", binaryPath+" ")
+
+	// Replace "touch" with "New-Item" on Windows as it's a common command, but not POSIX so not aliases by M$
+	// See https://mathieubuisson.github.io/powershell-linux-bash/ &
+	// http://web.cs.ucla.edu/~miryung/teaching/EE461L-Spring2012/labs/posix.html for more details
+	if runtime.GOOS == "windows" {
+		cmd = regexp.MustCompile(`^touch `).ReplaceAllString(cmd, `New-Item `)
+	}
+
+	return cmd, nil
+}
+
+// Merge the actionset defaults with the action config
+func actionGetCfg(actionSet types.ZarfComponentActionSet, a types.ZarfComponentAction) types.ZarfComponentActionDefaults {
+	cfg := actionSet.Defaults
+
+	if !a.Mute {
```

Comment:
There is a problem here with relying simply on boolean operations (at least for what I would expect).  Consider:

1. If `a.Mute` is not specified if will default to the zero value of the boolean in Go (`false`) and will always override the default `cfg.Mute`
2. If `a.Mute` is `true` it will never override the default `cfg.Mute` since `!true` is false

Reply:
This may need to be a nillable field.